src/dune_rules/compilation_context.ml

```diff
@@ -4,7 +4,7 @@ open Import
 module SC = Super_context
 
 module Includes = struct
-  type t = Command.Args.dynamic Command.Args.t Cm_kind.Dict.t
+  type t = Command.Args.without_targets Command.Args.t Cm_kind.Dict.t
```

Comment:
There are a few diffs like this, which are a bit surprising: does this mean that `t` actually never had any targets in it, so we could have used `static` instead of `dynamic` here?

Reply:
Well, it had no targets but it had `Dyn` so we had to use `dynamic`

Reply:
Ah, got it. So this shows the new encoding is clearer.